src/main/java/io/vertx/ext/mongo/CollationOptions.java

```diff
@@ -0,0 +1,380 @@
+package io.vertx.ext.mongo;
+
+import com.mongodb.client.model.*;
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.codegen.annotations.GenIgnore;
+import io.vertx.core.json.JsonObject;
+
+import java.util.Objects;
+
+/**
+ * Options used to configure collation options.
+ *
+ * @author <a href="mailto:christoph.spoerk@gmail.com">Christoph Spörk</a>
+ */
+@DataObject(generateConverter = true)
+public class CollationOptions {
+  public static final Boolean DEFAULT_CASE_LEVEL = false;
+  public static final CaseFirst DEFAULT_CASE_FIRST = CaseFirst.off;
+  public static final Alternate DEFAULT_ALTERNATE = Alternate.NON_IGNORABLE;
+  public static final Integer DEFAULT_STRENGTH = 3;
+  public static final Boolean DEFAULT_NUMERIC_ORDERING = false;
+  public static final Boolean DEFAULT_BACKWARDS = false;
+  public static final Boolean DEFAULT_NORMALIZATION = false;
+  public static final MaxVariable DEFAULT_MAX_VARIABLE = MaxVariable.punct;
+  String locale;
+  boolean caseLevel;
+  CaseFirst caseFirst;
+  int strength;
+  boolean numericOrdering;
+  String alternate;
+  MaxVariable maxVariable;
+  boolean backwards;
+  boolean normalization;
+
+  /**
+   * Default constructor for setting
+   */
+  public CollationOptions(String locale) {
```

Comment:
All Vert.x option classes must have at least 3 constructors: default (no param), copy and from json object.

Reply:
Unfortunately, I wasn't able to find the default locale set by mongo.
https://docs.mongodb.com/manual/reference/collation/
I fear that I'd change the intended behavior of setting a default locale without knowing that mongo also confirms to that.
Thus I decided to go for a mandatory locale, when using specific `CollationOptions`. @tsegismont do you know of a better solution or maybe which default locale I could set instead as a default one?

Reply:
If not locale is provided by the user, we should use the JVM's default locale, I think